tensorflow/python/keras/distribute/worker_training_state_test.py

```diff
@@ -17,46 +17,102 @@
 from tensorflow.python.distribute import multi_worker_test_base as test_base
 from tensorflow.python.framework import test_combinations as combinations
 from tensorflow.python.keras import callbacks
 from tensorflow.python.keras.distribute import multi_worker_testing_utils
+from tensorflow.python.keras.distribute import worker_training_state
 from tensorflow.python.lib.io import file_io
+from tensorflow.python.platform import googletest
 from tensorflow.python.platform import test
 
 
 class ModelCheckpointTest(test_base.IndependentWorkerTestBase,
                           parameterized.TestCase):
 
   @ds_combinations.generate(
-      combinations.combine(
-          mode=['graph'],
-          required_gpus=[0, 1],
-          file_format=['h5', 'tf'],
-          save_weights_only=[True, False]))
+      combinations.combine(mode=['graph'],
+                           required_gpus=[0, 1],
+                           file_format=['h5', 'tf'],
+                           save_weights_only=[True, False]))
   def testCheckpointExists(self, file_format, save_weights_only):
     with self.cached_session():
       train_ds, _ = multi_worker_testing_utils.mnist_synthetic_dataset(64, 2)
       model = multi_worker_testing_utils.get_mnist_model((28, 28, 1))
       saving_dir = self.get_temp_dir()
       saving_filepath = os.path.join(saving_dir, 'checkpoint.' + file_format)
       callbacks_list = [
-          callbacks.ModelCheckpoint(
-              filepath=saving_filepath, save_weights_only=save_weights_only)
+          callbacks.ModelCheckpoint(filepath=saving_filepath,
+                                    save_weights_only=save_weights_only)
       ]
       self.assertFalse(file_io.file_exists_v2(saving_filepath))
       model.fit(
           x=train_ds, epochs=2, steps_per_epoch=2, callbacks=callbacks_list)
       tf_saved_model_exists = file_io.file_exists_v2(saving_filepath)
       tf_weights_only_checkpoint_exists = file_io.file_exists_v2(
           saving_filepath + '.index')
-      self.assertTrue(
-          tf_saved_model_exists or tf_weights_only_checkpoint_exists)
+      self.assertTrue(tf_saved_model_exists or
+                      tf_weights_only_checkpoint_exists)
+
+
+def testWorkerTrainingState(cluster_spec, task_id, backup_dir):
+  #Set TFCONFIG
+  tf_config = {
+      'cluster': cluster_spec,
+      'task': {
+          'type': "worker",
+          'index': task_id
+      }
+  }
+  os.environ["TF_CONFIG"] = json.dumps(tf_config)
+  strategy = collective_all_reduce_strategy.CollectiveAllReduceStrategy()
+  train_ds, _ = multi_worker_testing_utils.mnist_synthetic_dataset(64, 2)
+  with strategy.scope():
+    model = multi_worker_testing_utils.get_mnist_model((28, 28, 1))
+
+  model._training_state = (worker_training_state.WorkerTrainingState(
+      model, backup_dir))
+  model._training_state.back_up(0)
+
+  len_backup = len(os.listdir(backup_dir))
+  assert len_backup > 0, "%d" % (len_backup)
```

Comment:
looks like it is possible for task0 to clean up the backup before others tasks reaching this line?